File: scripts/test_solutions/647_palindromic_substrings.js
var countSubstrings = function(s) {
    let count = 0;
    function expand(l, r) { while (l >= 0 && r < s.length && s[l] === s[r]) { count++; l--; r++; } }
    for (let i = 0; i < s.length; i++) { expand(i, i); expand(i, i + 1); }
    return count;
};

File: scripts/test_solutions/647_palindromic_substrings.cpp
class Solution {
public:
    int countSubstrings(string s) {
        int count = 0;
        int n = s.size();

        auto expand = [&](int l, int r) {
            while (l >= 0 && r < n && s[l] == s[r]) {
                count++;
                l--;
                r++;
            }
        };

        for (int i = 0; i < n; i++) {
            expand(i, i);
            expand(i, i + 1);
        }
        return count;
    }
};
